src/internal/data/metrics.go

```diff
@@ -15,3 +22,73 @@ func (cs CollectionStats) IsZero() bool {
 func (cs CollectionStats) String() string {
 	return cs.Details
 }
+
+type KindStats struct {
+	BytesRead     int64
+	ResourceCount int64
+}
+
+type ExportStats struct {
+	// data is kept private so that we can enforce atomic int updates
+	// TODO(meain): Should we be exposing details.ItemType?
+	data map[details.ItemType]KindStats
+}
+
+func (es *ExportStats) UpdateBytes(kind details.ItemType, bytesRead int64) {
+	if es.data == nil {
+		es.data = map[details.ItemType]KindStats{}
+	}
+
+	ks := es.data[kind]
+	atomic.AddInt64(&ks.BytesRead, bytesRead)
+	es.data[kind] = ks
+}
+
+func (es *ExportStats) UpdateResourceCount(kind details.ItemType) {
+	if es.data == nil {
+		es.data = map[details.ItemType]KindStats{}
+	}
+
+	ks := es.data[kind]
+	atomic.AddInt64(&ks.ResourceCount, 1)
+	es.data[kind] = ks
+}
+
+func (es *ExportStats) GetStats() map[details.ItemType]KindStats {
+	return es.data
+}
+
+type statsReader struct {
+	reader io.ReadCloser
+	kind   details.ItemType
```

Comment:
can we actually switch this to use path.CategoryType and/or path.ServiceType? That'll get us the benefit of being able to use the newly added `HumanString()` (at least for category, may need to add something similar for service) which helps keep output across various parts of the program consistent. We've also already got quite a bit of infrastructure that ensures those values can be compared/switched on as needed

If you need to make them into a map key (i.e. want both service and category) I suggest making a function to take a service/category pair and output a string or something. These are all in-memory values, so as long as the output is consistent for a given run of corso we can update the generated combined value as needed

Reply:
Ahh, forgot about `path.CategoryType`. Let me switch to that.